### PR TITLE
[luci] Revise SparsityParam

### PR DESCRIPTION
**Pull request by mhs4670go**
This commit revises sparsity parameter.

Related : #3889
Draft: #4394
ONE-DCO-1.0-Signed-off-by: seongwoo <mhs4670go@naver.com>

### CODE REVIEW

File: compiler/luci/lang/include/luci/IR/SparsityParam.h [hunk 1]
```diff
@@ -37,23 +39,186 @@ enum SparseIndexVectorType
   U8,
 };
 
-struct DimMetaData
+class SparseIndexVector
 {
+public:
+  SparseIndexVector() = default;
```
Comment (seanshpark) [inline compiler/luci/lang/include/luci/IR/SparsityParam.h:45]: I think default CTOR should be removed
```cxx
SparseIndexVector() = delete;
```
Current DTOR uses `_type` to deletes what was created

Comment (seanshpark) [reply compiler/luci/lang/include/luci/IR/SparsityParam.h:45]: Maybe remove copy CTOR too?

Comment (mhs4670go) [reply compiler/luci/lang/include/luci/IR/SparsityParam.h:45]:
```cxx
DimMetaData(DimensionType format, int32_t dense_size) : _format{format}, _dense_size{dense_size}
  {
    // DO NOTHING
  }
```
`DimMetaData` can be two types, e.g., dense, sparse. when it is sparse, `DimMetaData` calls above CTOR that calls default `SparseIndexVector` CTOR.

Comment (seanshpark) [reply compiler/luci/lang/include/luci/IR/SparsityParam.h:45]: Ah... didn't read the code carefully :)

**Review by seanshpark — APPROVED** [review]
LGTM

**Pull request merged**